src/app/proposals/shared/utils.ts

```diff
@@ -153,3 +155,57 @@ export const isChecksumValid = (value: string, chainId?: string) => {
     checksumAddress(value as Address, chainId ? Number(chainId) : undefined) === value
   )
 }
+
+type ProposalSource = 'DAO' | 'TALLY' | 'UNKNOWN'
+interface ParsedDescription {
+  name: string
+  description: string
+  source: ProposalSource
+}
+
+export const parseProposalDescription = (description: string): ParsedDescription => {
+  // Constants
+  const MAX_NAME_LENGTH = 100
```

Comment:
Move these constants outside the function
Also create a separate file for this function to avoid mixing this specific utility with the shared utils.

Reply:
This function is already very specific to proposals and used only in the `proposal/shared/util.ts` could you suggest where you reckon is the best place to keep it in, for me, having to create a new folder and/or file to keep it in is overkill, at least for now. I probably should remove the export tag on the function so it's very clear it's used only in the util file. WDYT?

Reply:
We can keep it as it is for now, and if in the future we need to re-use it then we'll just create the file along with the function

Always be future proofing